libkbfs/block_journal_test.go

```diff
@@ -274,6 +307,38 @@ func TestBlockJournalRemoveReferences(t *testing.T) {
 	require.Equal(t, serverHalf, half)
 }
 
+func TestBlockJournalDuplicateRemove(t *testing.T) {
+	ctx, cancel, tempdir, _, j := setupBlockJournalTest(t)
+	defer teardownBlockJournalTest(t, ctx, cancel, tempdir, j)
+
+	// Put the block.
+	data := []byte{1, 2, 3, 4}
+	bID, bCtx, _ := putBlockData(ctx, t, j, data)
+
+	require.Equal(t, int64(len(data)), j.getStoredBytes())
+	require.Equal(t, int64(len(data)), j.getUnflushedBytes())
+
+	// Remove the only reference to the block, then remove the
+	// block.
+	liveCounts, err := j.removeReferences(
+		ctx, kbfsblock.ContextMap{bID: {bCtx}})
+	require.NoError(t, err)
+	require.Equal(t, map[kbfsblock.ID]int{bID: 0}, liveCounts)
+	err = j.remove(bID)
+	require.NoError(t, err)
+
+	require.Equal(t, int64(0), j.getStoredBytes())
+	require.Equal(t, int64(len(data)), j.getUnflushedBytes())
```

Comment:
Yeah, this test is manually calling `remove()` (a private function) and so is violating the invariant. Normally, the flush should happen first, then the remove. Added a comment.